modules/objdetect/include/opencv2/objdetect/fractal_detector.hpp

```diff
@@ -0,0 +1,225 @@
+
+    CV_PROP_RW int id; ///< The ID of the marker.
+    CV_PROP_RW std::vector<cv::KeyPoint> keypts; ///< The keypoints of the marker. The first 4 are external corners.
+
+private:
+    CV_PROP_RW cv::Mat _M; ///< The matrix representation of the marker.
+    CV_PROP_RW cv::Mat _mask; ///< The mask of the marker.
+    CV_PROP_RW std::vector<int> _submarkers; ///< The IDs of the submarkers.
+};
+
+/**
+ * @brief Represents a set of fractal markers and their configurations.
+ * 
+ * This class manages a collection of fractal markers, including their configurations,
+ * relationships, and unit conversions.
+ */
+class CV_EXPORTS_W_SIMPLE FractalMarkerSet {
+public:
+    /**
+     * @brief Default constructor for FractalMarkerSet.
+     */
+    CV_WRAP FractalMarkerSet();
+
+    /**
+     * @brief Constructs a FractalMarkerSet with the given configuration.
+     * 
+     * @param config The configuration string (e.g., "FRACTAL_2L_6").
+     */
+    CV_WRAP FractalMarkerSet(std::string config);
+
+    /**
+     * @brief Converts the marker units to meters.
+     * 
+     * @param size The physical size of the marker in meters.
+     * @throws std::runtime_error If the markers are not expressed in pixels or normalized units.
+     */
+    CV_WRAP void convertToMeters(float size);
+
+    CV_PROP_RW std::map<int, FractalMarker> fractalMarkerCollection; ///< A map of marker IDs to FractalMarker objects.
+    CV_PROP_RW std::map<int, std::vector<int>> bits_ids; ///< A map of bit counts to marker IDs.
+    CV_PROP_RW int mInfoType; ///< Indicates the unit type (e.g., meters, pixels, or normalized).
+    CV_PROP_RW int idExternal; ///< The ID of the external marker.
+};
+
+
+/**
+ * @brief The FractalMarkerDetector class detects fractal markers in the images passed.
+ */
+class CV_EXPORTS_W_SIMPLE FractalMarkerDetector {
+public:
+    /**
+     * @brief Set parameters for the fractal marker detector.
+     * @param fractal_config Possible values (FRACTAL_2L_6, FRACTAL_3L_6, FRACTAL_4L_6, FRACTAL_5L_6).
+     * @param minInternalDistSq The minimum squared distance between internal points (default 150).
+     * @param markerSize The size of the marker in meters (optional).
+     */
+    CV_WRAP void setParams(const std::string& fractal_config, int minInternalDistSq = 150, float markerSize = -1);
+
+    /**
+     * @brief Detect fractal markers in the given image.
+     * @param img Input image.
+     * @return A vector of detected FractalMarker objects.
+     */
+    CV_WRAP std::vector<FractalMarker> detect(const cv::Mat& img);
+
+    /**
+     * @brief Detect fractal markers and retrieve 2D/3D correspondences.
+     * @param img Input image.
+     * @param p3d Output 3D points.
+     * @param p2d Output 2D points.
+     * @return A vector of detected FractalMarker objects.
+     */
+    CV_WRAP std::vector<FractalMarker> detect(const cv::Mat& img, std::vector<cv::Point3f>& p3d, std::vector<cv::Point2f>& p2d);
```

Comment:
You can use cv::OutputArray = cv::noArray() for the optional `p3d` and `p2d` and provide single method instead of overload. It's handled by bindings generator automatically.